x-pack/plugins/security_solution/public/common/components/navigation/security_side_nav/security_side_nav.tsx

```diff
@@ -65,7 +72,12 @@ const useFormatSideNavItem = (): FormatSideNavItems => {
       const formatDefaultItem = (navItem: NavLinkItem): DefaultSideNavItem => ({
         id: navItem.id,
         label: navItem.title,
-        ...getSecuritySolutionLinkProps({ deepLinkId: navItem.id }),
+        ...getSecuritySolutionLinkProps({
+          deepLinkId: navItem.id,
+          onClick: () => {
+            track(METRIC_TYPE.CLICK, `${TELEMETRY_EVENT.NAVIGATION}${navItem.id}`);
+          },
+        }),
```

Comment:
That's nice! There is another important `getSecuritySolutionLinkProps` call that we should track, at line 94 below.
However, I think that instead of adding all `onClick`s manually to the Security links, wouldn't it make more sense to encapsulate the "NAVIGATION" `track` calls inside the `getSecuritySolutionLinkProps` implementation? This way we'll track everything automatically, we can use the `deepLinkId` as the identifier.
Or are there some navigation clicks we don't want to track?

Reply:
Yeah, it'd be cleaner if we do the tracking in `getSecuritySolutionLinkProps `, but seems `useGetSecuritySolutionLinkProps` is called many places other than navigations, I'm worried the count wouldn't be correct if it includes the clicks from else where.